packages/trpc/src/procedures/recipes/getRecipes.spec.ts

```diff
@@ -20,30 +26,105 @@ describe("getRecipes", () => {
   it("returns a list of recipes given no filters", async () => {
     await prisma.recipe.create({
       data: {
-        userId: user.id,
+        ...getRecipeStats(user.id),
+      },
+    });
+
+    const response = await trpc.recipes.getRecipes.query({
+      limit: 3,
+      folder: "main",
+      orderBy: "title",
+      orderDirection: "asc",
+      offset: 0,
+    });
+
+    expect(typeof response.recipes[0].id).toBe("string");
+  });
+
+  it("filters by ratings", async () => {
+    await prisma.recipe.create({
+      data: {
+        ...getRecipeStats(user.id),
+        rating: 3,
+      },
+    });
+
+    await prisma.recipe.create({
+      data: {
+        ...getRecipeStats(user.id),
+        rating: 4,
+      },
+    });
+
+    const response = await trpc.recipes.getRecipes.query({
+      userIds: [user.id],
+      ratings: [3],
+      limit: 3,
+      folder: "main",
+      orderBy: "title",
+      orderDirection: "asc",
+      offset: 0,
+    });
+
+    expect(response.totalCount).toEqual(1);
+    expect(response.recipes[0].rating).toEqual(3);
+  });
+
+  it("filters by labels", async () => {
+    const label = await prisma.label.create({
+      data: {
         title: faker.string.alphanumeric(10),
-        description: faker.string.alphanumeric(10),
-        yield: faker.string.alphanumeric(10),
-        folder: "main",
-        activeTime: faker.string.alphanumeric(10),
-        totalTime: faker.string.alphanumeric(10),
-        source: faker.string.alphanumeric(10),
-        url: faker.string.alphanumeric(10),
-        notes: faker.string.alphanumeric(10),
-        ingredients: faker.string.alphanumeric(10),
-        instructions: faker.string.alphanumeric(10),
+        userId: user.id,
+        labelGroupId: null,
+      },
+    });
+
+    await prisma.recipe.create({
+      data: {
+        ...getRecipeStats(user.id),
+        title: "myTitle",
+        recipeLabels: {
+          createMany: {
+            data: [{ labelId: label.id }],
+          },
+        },
         rating: faker.number.int({ min: 1, max: 5 }),
       },
     });
 
+    await prisma.recipe.create({
+      data: {
+        ...getRecipeStats(user.id),
+      },
+    });
+
     const response = await trpc.recipes.getRecipes.query({
+      userIds: [user.id],
+      labels: [label.title],
       limit: 3,
       folder: "main",
       orderBy: "title",
       orderDirection: "asc",
       offset: 0,
     });
 
-    expect(typeof response.recipes[0].id).toBe("string");
+    expect(response.totalCount).toEqual(1);
+    expect(response.recipes[0].title).toEqual("myTitle");
+  });
+});
+
+describe("getRecipes - error cases", () => {
```

Comment:
[non-blocking] Generally you'd have your tests organized in describe blocks as follows:

- getRecipes
  - success
    - test1
    - test2
  - error
    - test3
    - test4